src/main/java/org/opensrp/web/rest/UploadController.java

```diff
@@ -0,0 +1,378 @@
+		event.setEventType(eventType);
+		event.setType("Event");
+	}
+
+	private void assignClientUniqueID(Client client, String uniqueIDKey, UniqueIDProvider uniqueIDProvider) {
+		if (StringUtils.isBlank(client.getIdentifier(uniqueIDKey))) {
+			Map<String, String> identifiers = client.getIdentifiers();
+			if (identifiers == null)
+				identifiers = new LinkedHashMap<>();
+
+			identifiers.put(uniqueIDKey, uniqueIDProvider.getNewUniqueID());
+
+			client.setIdentifiers(identifiers);
+		}
+	}
+
+	@RequestMapping(value = "/history", method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<String> getUploadHistory(HttpServletRequest request) {
+		Integer batchSize = getIntegerFilter(BATCH_SIZE, request);
+		if (batchSize == null)
+			batchSize = 50;
+
+		Integer offset = getIntegerFilter(OFFSET, request);
+		if (offset == null)
+			offset = 0;
+
+		String providerID = getCurrentUser();
+
+		List<UploadBean> uploadBeans = multimediaRepository.getByProviderID(providerID, FILE_CATEGORY, offset, batchSize)
+				.stream()
+				.map(multimedia -> {
+					UploadBean uploadBean = new UploadBean();
+					uploadBean.setFileName(multimedia.getOriginalFileName());
+					uploadBean.setIdentifier(multimedia.getCaseId());
+					uploadBean.setProviderID(multimedia.getProviderId());
+					uploadBean.setUploadDate(multimedia.getDateUploaded());
+					uploadBean.setUrl(multimedia.getCaseId() + "." + FILE_CATEGORY);
+					return uploadBean;
+				})
+				.collect(Collectors.toList());
+
+		return new ResponseEntity<>(
+				gson.toJson(uploadBeans),
+				RestUtils.getJSONUTF8Headers(), HttpStatus.OK);
+	}
+
+	@RequestMapping(value = "/template", method = RequestMethod.GET)
+	public void getUploadTemplate(@RequestParam("event_name") String eventName, HttpServletRequest request,
+			HttpServletResponse response) {
+		String locationID = getStringFilter(LOCATION_ID, request);
+
+		String csvFileName = eventName.replace(" ", "").toLowerCase() + ".csv";
+		response.setContentType("text/csv");
+		String headerKey = "Content-Disposition";
+		String headerValue = String.format("attachment; filename=\"%s\"",
+				csvFileName);
+		response.setHeader(headerKey, headerValue);
+
+		List<CSVRowConfig> configs = uploadService.getCSVConfig(eventName);
+
+		String[] HEADERS = configs.stream().map(CSVRowConfig::getColumnName)
+				.toArray(String[]::new);
+
+		List<String> fieldMappings = configs.stream().map(CSVRowConfig::getFieldMapping)
+				.collect(Collectors.toList());
+
+		List<Client> clients = clientService.findAllByAttribute(DEFAULT_RESIDENCE, locationID);
+
+		try (CSVPrinter printer = new CSVPrinter(response.getWriter(), CSVFormat.DEFAULT
+				.withHeader(HEADERS))) {
+			for (Client client : clients) {
+
+				JSONObject jsonObject = new JSONObject();
+				jsonObject.put("client", new JSONObject(gson.toJson(client)));
+				printer.printRecord(JSONCSVUtil.jsonToString(jsonObject, fieldMappings));
+			}
+		}
+		catch (IOException e) {
```

Comment:
Remove this so that its caught in global exception handler or return an error code